src/fidesops/service/connectors/saas_query_config.py

```diff
@@ -113,12 +123,65 @@ def generate_update_stmt(
                     f"<{param.name}>",
                     pydash.get(param_values, param.references[0].field),
                 )
+            elif param.type == "body":
+                custom_body = SaaSQueryConfig._build_request_body(
+                    custom_body,
+                    path,
+                    param.name,
+                    param.default_value,
+                    pydash.get(param_values, param.references[0].field)
+                    if param.references
+                    else None,
+                    pydash.get(param_values, param.identity)
+                    if param.identity
+                    else None,
+                )
 
         logger.info(f"Populated request params for {current_request.path}")
 
         update_value_map: Dict[str, Any] = self.update_value_map(row, policy, request)
         body: Dict[str, Any] = unflatten_dict(update_value_map)
-        return "PUT", path, params, json.dumps(body)
+        if custom_body:
+            # removes outer {} wrapper from body for greater flexibility in custom body config
+            custom_body = custom_body.replace(
+                "<masked_object_fields>", json.dumps(body)[1:-1]
+            )
+        return (
+            "PUT",
+            path,
+            params,
+            json.dumps(json.loads(custom_body) if custom_body else body),
+        )
+
+    @staticmethod
+    def _build_request_body(  # pylint: disable=R0913
+        custom_body: Optional[str],
```

Comment:
If this field is optional we should provide a default and move it to the end of the kwargs list